projects/ngx-d3timeline/src/lib/store/store.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable } from '@angular/core';
+import { ReplaySubject } from 'rxjs';
+import { ActionTypes } from './actions';
+import { shareReplay, scan } from 'rxjs/operators';
+import { initialState } from './state';
+import { State } from './state';
+import { TimelineView } from '../view/timeline-view';
+import { Actions } from './actions';
+import { ScaleService } from '../scale.service';
+import { OptionsService } from '../options.service';
+
+@Injectable({ providedIn: 'root' })
+export class Store {
+  private readonly replayBuffer = 100;
+  private actionsSubject = new ReplaySubject<Actions>(this.replayBuffer);
+
+  constructor(
+    private scaleService: ScaleService,
+    private optionsService: OptionsService
+  ) {}
+
+  state$ = this.actionsSubject.pipe(
+    scan(this.reducer.bind(this), initialState),
+    shareReplay()
+  );
+
+  dispatch(action: Actions) {
+    this.actionsSubject.next(action);
+  }
+
+  private reducer(state: State, action: Actions): State {
+    switch (action.type) {
+      case ActionTypes.DataChanged: {
+        return this.recomputeStateAndScales(state, { data: action.payload });
+      }
+
+      case ActionTypes.OrientationChanged: {
+        return this.recomputeStateAndScales(state, {
+          axisOrientations: this.optionsService.setAxisOrientations(
+            action.payload
+          )
+        });
+      }
+
+      case ActionTypes.ViewChanged: {
+        return this.recomputeStateAndScales(state, {
+          view: new TimelineView(action.payload)
+        });
+      }
+
+      case ActionTypes.ZoomEvent: {
+        return {
+          ...state,
+          timeScale: this.scaleService.rescaleTime(state, action.payload)
+        };
+      }
+
+      default: {
+        return state;
+      }
+    }
+  }
+
+  private recomputeStateAndScales(state: State, bar: Partial<State>) {
```

Comment:
Method name change as its not really 'recomputing' the state + change of `bar` parameter name.

A bit uneasy with this method doing two things.

```suggestion
  private patchStateAndRecomputeScales(state: State, patch: Partial<State>) {
```
OR
```suggestion
  private patchStateAndUpdateScales(state: State, patch: Partial<State>) {
```

Reply:
> A bit uneasy with this method doing two things.

Agreed have an idea to sort it, will do it in another issue